### PR TITLE
Add new parameter to createLoadBalancerRule API

### PR DESCRIPTION
**Pull request by JoaoJandre**
### Description

ACL rules are currently not enforced when a Load Balancer is used. When created, ACL rules are implemented in the VR in the FORWARD chain. In order for the LB to work only through the configured IPs, it is necessary to configure the rule in the INPUT chain. However, since all traffic into the VR is being allowed, only forwarding is blocked to the unconfigured IPs.

To work around this situation, it is possible to configure LB rules via `createLoadBalancerRule` API. However, the `cidrlist` parameter used to configure the CIDRs that can access the load balancer, had been deprecated in favor of the firewall rule implementation, which is not intuitive and does not support VRs in VPCs. Therefore, the parameter was reimplemented to allow users to create rules in the Load Balancer backend that restrict the access according to the source IP.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [X] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?
New unit tests were created. Alongside that, the changes were manually tested in a local lab. By using the `createLoadBalancerRule` API with the `cidrlist` parameter to restrict network traffic.

### CODE REVIEW

**Comment by weizhouapache** [comment]
@JoaoJandre 
there are many parameters supported by haproxy but not supported in cloudstack.
I suggest you to collaborate with @ravening on https://github.com/apache/cloudstack/pull/5799

**Comment by JoaoJandre** [comment]
> @JoaoJandre there are many parameters supported by haproxy but not supported in cloudstack. I suggest you to collaborate with @ravening on #5799

I would rather move on with this PR only, as it is re-adding a feature that was deprecated/removed without a proper alternative. This PR is small, concise, unit tested, and ready to be used in production. The alternative you pointed out is a huge PR that looks a bit too complicated. Therefore, I would rather move on with this small addition to restore the API workings.
Anyways, I can note that PR here, and if we ever need such a featureset, we might be able to help there.

**Comment by GutoVeronezi** [comment]
@weizhouapache, PR #5799 has being developed through 2 years from now (originally developed in PR #4141). It passed through 4.15, 4.16, and 4.17, and we do not have guarantee that will be merged in 4.18.  Also, there is a lot of changes in #5799, which would require a good documentation and a lot of tests. Therefore, I would rather go with these small changes instead of blocking the feature for something we do not have guarantee.

**Comment by weizhouapache** [comment]
> @weizhouapache, PR #5799 has being developed through 2 years from now (originally developed in PR #4141). It passed through 4.15, 4.16, and 4.17, and we do not have guarantee that will be merged in 4.18. Also, there is a lot of changes in #5799, which would require a good documentation and a lot of tests. Therefore, I would rather go with these small changes instead of blocking the feature for something we do not have guarantee.

@GutoVeronezi @JoaoJandre 
as far as I know, #5799 is not beeing developed, tt has been deployed in their production for around two years. 
It is ready for review and test. The reason it is not merged is, nobody in cloudstack community has reviewed and tested it.
I hope you guys can review/test it and collaborate with @ravening @soreana on merging #5799 and changes for cidrlist into 4.18

**Comment by RodrigoDLopez** [comment]
based on tests performed in our labs and code reviews. Lgtm.

**Comment by weizhouapache** [comment]
> @weizhouapache I have a suggestion for #5799. We can test it gradually, like what we did when we developed it. We can have a branch named `HAproxy-LoadBalancer` in the community. I will port all the changes from #5799 to `HAproxy-LoadBalancer` one by one. When satisfied with its quality, we can merge it to the `main` branch. That way, it would be far easier for community members to test it.
> 
> This feature could be part of it, too.

@soreana 
Each PR can be merged only if there are at least 2 approvals and all smoke tests pass.
Most users use official release. If someone want to test a feature, they can build and test the specific branch, it is not needed to merge it into main.

**Comment by JoaoJandre** [comment]
Hey @DaanHoogland could you review and run the tests?

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Review by DaanHoogland — APPROVED** [review]
clgtm needs testing

File: api/src/main/java/org/apache/cloudstack/api/command/user/loadbalancer/CreateLoadBalancerRuleCmd.java [hunk 1]
```diff
@@ -107,7 +107,8 @@ public class CreateLoadBalancerRuleCmd extends BaseAsyncCreateCmd /*implements L
     @Parameter(name = ApiConstants.DOMAIN_ID, type = CommandType.UUID, entityType = DomainResponse.class, description = "the domain ID associated with the load balancer")
     private Long domainId;
 
-    @Parameter(name = ApiConstants.CIDR_LIST, type = CommandType.LIST, collectionType = CommandType.STRING, description = "the CIDR list to forward traffic from. Multiple entries must be separated by a single comma character (,). This parameter is deprecated. Do not use.")
+    @Parameter(name = ApiConstants.CIDR_LIST, type = CommandType.LIST, collectionType = CommandType.STRING, description = "the CIDR list to allow traffic, all other CIDRs"
```
[the remark this answers is not in the document]
Comment (JoaoJandre) [reply api/src/main/java/org/apache/cloudstack/api/command/user/loadbalancer/CreateLoadBalancerRuleCmd.java]: No problem, thanks for the review :)

**Pull request merged**